src/psyclone/gocean1p0.py

```diff
@@ -3141,6 +3170,27 @@ def gen_code(self, parent):
 
         # TODO 886: Currently only stencils of depth 1 are accepted by this
         # API, so the HaloExchange is hardcoded to depth 1.
+
+        if self.ancestor(InvokeSchedule).opencl:
+            # If the dependency is a Kernel add a barrier if the previous
+            # kernels was dispatched in a different command queue
+            dependency = self.backward_dependence()
```

Comment:
Maybe but these are 2 different nodes and I am unsure where I would place the common functionality without feeling out of place. However, this should be moved eventually to the PSyIR OpenCL transformation, so I propose give another look at removing the duplication at #1134, because then it can be a private function of the transformation.

Reply:
OK, I'm happy with that.